src/vconsole/vconsole-setup.c

```diff
@@ -93,14 +89,14 @@ static bool is_settable(int *fd, int idx) {
          *
          * http://lists.freedesktop.org/archives/systemd-devel/2013-February/008573.html
          */
-        if (r != 0 || (curr_mode != K_XLATE && curr_mode != K_UNICODE)) {
-                if (*fd < 0)
+        if (r != 0 || !IN_SET(curr_mode, K_XLATE, K_UNICODE)) {
+                /* close terminal only if we opened it in this function */
+                if (fd < 0)
                         (void) close(fdt);
-                return false;
+                return r < 0 ? r : -1;
```

Comment:
We don't return "fake" errors in systemd code. The error should always be a negative errno code, and -1 means EPERM.